### DISCUSSION TITLE
Documentation on the development setup with qtcreator and docker

### DISCUSSION THREAD

**Discussion by raju-bitter** [slotThe/RemarkableLamyEraser#14] [Q&A]
Is there any documentation on how to set up Qtcreator running in the Docker instance to modify the source and build the project? This is what I'm seeing after opening _Recent Projects -> /mnt/RemarkableLamyEraser/RemarkableLamyEraser.pro_.
![image](https://github.com/slotThe/RemarkableLamyEraser/assets/296926/56eb1f39-5a5f-4dde-82fe-6ffeea025b32)

Had to allow X forwarding for root by running this command:
`$ xhost +si:localuser:root`

Then Docker launches with Qtcreator. I'm on a 5k display, and the QtCreator UI elements as well as text look tiny. What do I need to enable scaling of the UI to match the high DPI resolution?

Maybe @slotThe or @rish987 can help me with that. It should be documented for other people interested in doing customizations as well.

**Comment by slotThe** [comment]
The [README](https://github.com/slotThe/RemarkableLamyEraser#developing) has a short paragraph about it; I don't know whether UI scaling works this way.

In your picture you can click on "Configure Project", and then `C-b` should build the project as normal.

**Comment by rish987** [comment]
Also, if you want to use a more "standard" editor like emacs, neovim etc., it's also possible to just run `qmake` from the root directory, which will generate the Makefile, and then use a tool like [compiledb](https://github.com/nickdiego/compiledb) to generate the `compile_commands.json` file that `clangd` will need for LSP things to work. That's how I have been developing (perhaps it's worth mentioning in the README).

**Reply by slotThe** [reply]
> (perhaps it's worth mentioning in the README).

I think so, yes!
